src/pymatgen/io/vasp/inputs.py

```diff
@@ -1424,7 +1424,7 @@ def automatic_density_by_lengths(
         lattice = structure.lattice
 
         abc = lattice.abc
-        num_div: Tuple3Ints = tuple(np.ceil(ld / abc[idx]) for idx, ld in enumerate(length_densities))
+        num_div: Tuple3Ints = tuple(math.ceil(ld / abc[idx]) for idx, ld in enumerate(length_densities))
```

Comment:
Need confirmation here: `np.ceil` would return `float`, I think it should be `int` instead in this context.

Reply:
lgtm